packages/manager-react-components/src/hooks/services/hooks/useServiceDetails.ts

```diff
@@ -0,0 +1,35 @@
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import {
+  getResourceServiceId,
+  getResourceServiceIdQueryKey,
+  getServiceDetails,
+} from '../api';
+import { ServiceDetails } from '../services.type';
+
+export const serviceDetailsQueryKey = ['service-details'];
+
+export type UseServiceDetailsParams = {
+  queryKey?: string[];
+  resourceName: string;
+};
+
+export const useServiceDetails = ({
+  queryKey = serviceDetailsQueryKey,
+  resourceName,
+}: UseServiceDetailsParams) => {
+  const queryClient = useQueryClient();
+  return useQuery<ApiResponse<ServiceDetails>, ApiError>({
+    queryKey,
+    queryFn: async () => {
+      const { data } = await queryClient.fetchQuery<
+        ApiResponse<number[]>,
+        ApiError
+      >({
+        queryKey: getResourceServiceIdQueryKey({ resourceName }),
+        queryFn: () => getResourceServiceId({ resourceName }),
+      });
+      return getServiceDetails(data[0]);
+    },
+  });
+};
```

Comment:
For readability, may be it's better to use 2 useQuery with this second query a disabled key for wait data[0] ?

Reply:
There are 2 problems :
- you double the code to maintain for the same result
- if I return the result of the second useQuery, it won't be loading until the first query returns. Or I could return the isLoading of the first query. Same thing for the error management. Why bother when I could just use one useQuery ?

Reply:
I think is more readable and more reusable. This avoids rewrite useResource if we needs and recreate different `queryKey`. 

We can update serviceDetails query key too

```ts
export const useResource = ({
  resourceName,
}: UseServiceDetailsParams) => {
  return useQuery<ApiResponse<Resource>, ApiError>({
    queryKey: getResourceServiceIdQueryKey({ resourceName }),
    queryFn: getResourceServiceId({ resourceName }),
  });
};

export const useServiceDetails = ({
  resourceName,
}: UseServiceDetailsParams) => {
  const {resourceData, isSuccess} = useResource(resourceName)

  return useQuery<ApiResponse<ServiceDetails>, ApiError>({
    queryKey: generateMyKey(resourceName, resourceData[0]),
    queryFn: getServiceDetails(resourceData[0]),
    enabled: isSuccess && resourceData,
  });

  return queryClient
};

```

Reply:
The code is in two function but less complex. So easier to maintain
And for isFetching and isError

 ```ts
export const useServiceDetails = ({
  resourceName,
}: UseServiceDetailsParams) => {
  const {resourceData, isSuccess} = useResource(resourceName)

  const { isError, error, isSuccess, isFetching, isPending, ...rest} =  useQuery<ApiResponse<ServiceDetails>, ApiError>({
    queryKey: generateMyKey(resourceName, resourceData[0]),
    queryFn: getServiceDetails(resourceData[0]),
    enabled: isSuccess && resourceData,
  });

  return { isError: isError || resourceData.isError, error: error || resourceData.error, isSuccess: isSuccess && resourceData.isSuccess, isFetching: isFetching || resourceData.isFetching, isPending: isPending: resourceData.isPending, ...rest }
};
```

I think we can separate this merge mechanism in a useful utils to simplify this function

Reply:
It's the principle of Single responsibility of SOLID to improve maintainable code and avoid duplicate code.

I'm agree on the first view it's not very complex and easier to write this function with the good finality.

But I think in this case this principle can be good in the time.

Reply:
"Single responsibility" != "Single operation"
The hook has only one responsibility : querying the service details data from the resource name.

The only benefit I see from extracting the part that query the service id from the resource name would be reusability (since it doesn't improve maintainability or performance as stated previously) but I don't see any use-case where we use the service id separately.

Do you see any other benefit ?